### PR TITLE
Updated react-responsive types to v3.0

### PR DESCRIPTION
**Pull request by alechill**
Also addresses https://github.com/DefinitelyTyped/DefinitelyTyped/issues/20656

https://github.com/contra/react-responsive

- types was completely broken by v3 release due to change in default export
- as mentioned in issue https://github.com/DefinitelyTyped/DefinitelyTyped/issues/20656 and https://github.com/contra/react-responsive/issues/126 many features are missing from the types, so have added here

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/react-responsive/index.d.ts
Comment (xaviergonz) [inline types/react-responsive/index.d.ts]: why not just the keys themselves as strings and then remove the interface? I don't think it's used anywhere

**Review by xaviergonz — APPROVED** [review]

File: types/react-responsive/index.d.ts
Comment (xaviergonz) [inline types/react-responsive/index.d.ts]: Javier actually :)

Comment (alechill) [reply types/react-responsive/index.d.ts]: 👍

File: types/react-responsive/index.d.ts
Comment (alechill) [reply types/react-responsive/index.d.ts]: It is used to extend the queryable properties for `MediaQueryAllQueryable` => can be applied to component and to the standalone `toQuery` helper

Allows for `<MediaQuery screen={true} handheld={true}>Some handheld device content</MediaQuery>`

**Comment by sheetalkamat** [comment]
You need to copy old definitions into v1 and update tsconfig correctly

**Comment by RyanCavanaugh** [comment]
Approved by third party. PR appears ready to merge pending review by a maintainer.

**Pull request merged**
